src/m2n/PointToPointCommunication.hpp

```diff
@@ -114,6 +141,25 @@ class PointToPointCommunication : public DistributedCommunication
    */
   std::vector<Mapping> _mappings;
 
+   /**
+   * @brief this data structure is used to store m2n communication information for the 1 step of 
+   *        bounding box initialization. It stores:
+   *        1. global remote process rank;
+   *        2. communication object (provides point-to-point communication routines).
+   *        3. Appropriatly sized buffer to receive elements
+   */
+  struct ConnectionData {
+    int                   remoteRank;
+    com::PtrCommunication communication;
+    com::PtrRequest       request;
```

Comment:
Why actually do we need the request here?

Reply:
The request is only needed later for the actual communication. But OK to keep it here.